pkg/infrastructure/aws/clusterapi/ignition.go

```diff
@@ -0,0 +1,71 @@
+package clusterapi
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/sirupsen/logrus"
+	capa "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+	"github.com/openshift/installer/pkg/infrastructure/clusterapi"
+	awstypes "github.com/openshift/installer/pkg/types/aws"
+	"github.com/openshift/installer/pkg/types/dns"
+)
+
+func editIgnition(ctx context.Context, in clusterapi.IgnitionInput) ([]byte, error) {
+	if in.InstallConfig.Config.AWS.UserProvisionedDNS != dns.UserProvisionedDNSEnabled {
+		return in.BootstrapIgnData, nil
+	}
+
+	awsCluster := &capa.AWSCluster{}
+	key := k8sClient.ObjectKey{
+		Name:      in.InfraID,
+		Namespace: capiutils.Namespace,
+	}
+	if err := in.Client.Get(ctx, key, awsCluster); err != nil {
+		return nil, fmt.Errorf("failed to get AWSCluster: %w", err)
+	}
+
+	awsSession, err := in.InstallConfig.AWS.Session(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get aws session: %w", err)
+	}
+
+	securityGroupIDs := []*string{}
+	for _, securityGroup := range awsCluster.Status.Network.SecurityGroups {
+		logrus.Warnf("Found Security Group ID: %s", securityGroup.ID)
+		securityGroupIDs = append(securityGroupIDs, aws.String(securityGroup.ID))
+	}
+	nicInput := ec2.DescribeNetworkInterfacesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("group-id"),
+				Values: securityGroupIDs,
+			},
+		},
+	}
+	nicOutput, err := ec2.New(awsSession).DescribeNetworkInterfacesWithContext(ctx, &nicInput)
+	if err != nil {
+		return nil, fmt.Errorf("failed to describe network interfaces: %w", err)
+	}
+
+	// The only network interfaces existing at this stage are those from the load balancers.
+	// If this stage is executed after control plane nodes are provisioned, there may be
+	// other network interfaces available.
+	publicIPAddresses := []string{}
+	privateIPAddresses := []string{}
+	for _, nic := range nicOutput.NetworkInterfaces {
+		if nic.Association != nil && nic.Association.PublicIp != nil {
+			logrus.Debugf("found public IP address associated with %s", *nic.Description)
```

Comment:
Since we're logging, maybe we want to include the addr itself?
```suggestion
			logrus.Debugf("found public IP address %s associated with %s", *nic.Association.PublicIp, *nic.Description)
```